consensus/core/src/tests/base_committer_tests.rs

```diff
@@ -0,0 +1,732 @@
+            .set_ancestors(references_without_leader_round_wave_4.clone())
+            .set_transactions(vec![Transaction::new(vec![2])])
+            .build(),
+    );
+    dag_state
+        .write()
+        .accept_block(byzantine_block_c13_2.clone());
+
+    let byzantine_block_c13_3 = VerifiedBlock::new_for_test(
+        TestBlock::new(13, 2)
+            .set_ancestors(references_without_leader_round_wave_4)
+            .set_transactions(vec![Transaction::new(vec![3])])
+            .build(),
+    );
+    dag_state
+        .write()
+        .accept_block(byzantine_block_c13_3.clone());
+
+    // Ancestors of decision blocks in round 14 should include multiple byzantine non-votes C13
+    // but there are enough good votes to prevent a skip. Additionally only one of the non-votes
+    // per authority should be counted so we should not skip leader A12.
+    let decison_block_a14 = VerifiedBlock::new_for_test(
+        TestBlock::new(14, 0)
+            .set_ancestors(good_references_voting_round_wave_4.clone())
+            .build(),
+    );
+    dag_state.write().accept_block(decison_block_a14.clone());
+
+    let good_references_voting_round_wave_4_without_c13 = good_references_voting_round_wave_4
+        .into_iter()
+        .filter(|r| r.author != AuthorityIndex::new_for_test(2))
+        .collect::<Vec<_>>();
+
+    let decison_block_b14 = VerifiedBlock::new_for_test(
+        TestBlock::new(14, 1)
+            .set_ancestors(
+                good_references_voting_round_wave_4_without_c13
+                    .iter()
+                    .cloned()
+                    .chain(std::iter::once(byzantine_block_c13_1.reference()))
+                    .collect(),
+            )
+            .build(),
+    );
+    dag_state.write().accept_block(decison_block_b14.clone());
+
+    let decison_block_c14 = VerifiedBlock::new_for_test(
+        TestBlock::new(14, 2)
+            .set_ancestors(
+                good_references_voting_round_wave_4_without_c13
+                    .iter()
+                    .cloned()
+                    .chain(std::iter::once(byzantine_block_c13_2.reference()))
+                    .collect(),
+            )
+            .build(),
+    );
+    dag_state.write().accept_block(decison_block_c14.clone());
+
+    let decison_block_d14 = VerifiedBlock::new_for_test(
+        TestBlock::new(14, 3)
+            .set_ancestors(
+                good_references_voting_round_wave_4_without_c13
+                    .iter()
+                    .cloned()
+                    .chain(std::iter::once(byzantine_block_c13_3.reference()))
+                    .collect(),
+            )
+            .build(),
+    );
+    dag_state.write().accept_block(decison_block_d14.clone());
+
+    // DagState Update:
+    // - We have A13, B13, D13 & C13 as good votes in the voting round of wave 4
+    // - We have 3 byzantine C13 nonvotes that we received as ancestors from decision
+    //   round blocks from B, C, & D.
```

Comment:
Great test , I think we might also want to test the case - we could do in a future PR - where a byzantine node is producing multiple voting blocks for a round where the other validators don't vote for the leader.